src/Symfony/Component/DependencyInjection/Secrets/JweHandler.php

```diff
@@ -0,0 +1,291 @@
+            throw new InvalidArgumentException(sprintf(
+                'publicKeyLocation %s must point to an existing %u byte file',
+                $this->publicKeyLocation,
+                SODIUM_CRYPTO_BOX_SECRETKEYBYTES
+            ));
+        }
+
+        if ($decryptRequired && !isset($this->privateKeyLocation)) {
+            throw new InvalidArgumentException('encrypted_secrets must be configured with a private_key_file location to read secrets.');
+        }
+
+        if ($decryptRequired && !$this->isValidKeyLocation($this->privateKeyLocation)) {
+            throw new InvalidArgumentException(sprintf(
+                'the configuration for encrypted_secrets.private_key_file (%s) must point to an existing %u byte file',
+                $this->privateKeyLocation,
+                SODIUM_CRYPTO_BOX_SECRETKEYBYTES
+            ));
+        }
+    }
+
+    /**
+     * Decrypts all populated secrets and writes the plaintext to the provided $fileLocation.
+     */
+    public function writePlaintext(string $fileLocation): void
+    {
+        $this->validateConfig($decryptRequired = true);
+        $plaintextSecrets = array();
+        foreach (array_keys($this->secrets) as $key) {
+            $plaintextSecrets[$key] = $this->decrypt($key);
+        }
+
+        $this->writeFile($fileLocation, json_encode($plaintextSecrets, self::JSON_ENCODE_OPTIONS));
+    }
+
+    /**
+     * helper method to format a sodium crypto box keypair from the configured public and private key values.
+     *
+     * @return string
+     */
+    private function collectKeyPair()
+    {
+        return sodium_crypto_box_keypair_from_secretkey_and_publickey(
+            $this->getPrivateKey(),
+            $this->getPublicKey()
+        );
+    }
+
+    private function getPrivateKey(): string
+    {
+        return $this->readKey($this->privateKeyLocation);
+    }
+
+    private function getPublicKey(): string
+    {
+        return $this->readKey($this->publicKeyLocation);
+    }
+
+    private function isValidKeyLocation(string $filePath): bool
+    {
+        return $this->isValidSecretsFile($filePath) &&
+            (SODIUM_CRYPTO_BOX_SECRETKEYBYTES === filesize($filePath));
+    }
+
+    private function isValidSecretsFile(string $filePath): bool
+    {
+        return stream_is_local($filePath) && is_readable($filePath);
+    }
+
+    /**
+     * @return array|false
+     */
+    private function populateSecrets()
+    {
+        return $this->secrets = $this->readSecrets($this->secretsLocation);
+    }
+
+    private function readKey(string $keyLocation)
+    {
+        return file_get_contents($keyLocation);
+    }
+
+    /**
+     * returns the decoded json secrets from $secretsLocation or false if the file does not contain valid json.
+     *
+     * @return array|false
+     */
+    private function readSecrets(string $secretsLocation)
+    {
+        $rawSecrets = trim(file_get_contents($secretsLocation));
```

Comment:
Any reason to trim this?

Reply:
no good catch this should be cleaned up on decode. removing